### PR TITLE
Spec GROQ Selectors & Diff namespace

### PR DESCRIPTION
**Pull request by sgulseth**

### CODE REVIEW

File: spec/Section 10 -- Functions.md [hunk 1]
```diff
@@ -317,3 +317,55 @@ dateTime::nowValidate(args):
...
+diff::changedAny(args, scope):
+
+- Let {lhs} be the first element of {args}.
+- Let {rhs} be the second element of {args}.
+- Let {selector} be the third element of {args}.
+- Let {before} be the result of {Evaluate(lhs, scope)}.
+- Let {after} be the result of {Evaluate(rhs, scope)}.
+- Let {evaluatedSelector} be the result of {EvaluateSelector(selector, scope)}
+- Let {search} be the result of {SelectorSearch(evaluatedSelector, before, scope)}
+- Let {diff} be the result of comparative difference between {before} and {after}
```
Comment (atombender) [inline spec/Section 10 -- Functions.md:338]: Maybe explain what "comparative difference" means?

File: spec/Section 2 -- Syntax.md [hunk 2]
```diff
@@ -73,6 +73,17 @@ Identifiers are used to name entities such as parameters, attributes and functio
 
 Identifier : /[A-Za-z\_][a-za-z_0-9]\*/
 
+## Selector
+
+To select a subset of fields in a document GROQ implements selectors. A selector can either be an [Identifier](#sec-Identifier) or a tuple of Identifiers.
```
Comment (atombender) [inline spec/Section 2 -- Syntax.md:78]: Maybe clarify that a selector isn't really a distinct type of expression, but that they're intended as a subset for the diff functions to use, and that the evaluation happens during diffing.

**Review by atombender — APPROVED** [review]
Are we sure that we want this to be part of the spec? It seems very Sanity webhook-specific.

File: spec/Section 10 -- Functions.md (hunk 1 above)
Comment (sgulseth) [reply spec/Section 10 -- Functions.md:338]: Agreed async that we can leave this as is and make it clearer later

**Review by atombender — APPROVED** [review]

**Pull request merged**